libminifi/src/c2/C2Agent.cpp

```diff
@@ -506,13 +527,16 @@ void C2Agent::handle_describe(const C2ContentResponse &resp) {
       }
 
       std::vector<std::shared_ptr<state::response::ResponseNode>> metrics_vec;
-
-      reporter->getResponseNodes(metrics_vec, metric_class_id);
       C2Payload response(Operation::ACKNOWLEDGE, resp.ident, false, true);
       response.setLabel("metrics");
+
+      C2Payload metrics(Operation::ACKNOWLEDGE, resp.ident, false, true);
```

Comment:
Why are two layers of `C2Payload` needed with the same label? "jstack" has only one to which the children containing the real data is added.
"configuration" is rewritten here to have a "configurationOptions" node to which an other "configurationOptions" node is added to which the children are added.
"manifest" is written with the same extra layer.

Reply:
That is because of the way C2Payload and C2ContentResponse classes are structured and the way we serializeJson payload. This needs some cleanup and might require some larger effort. Will take a note and open a Jira ticket to simplify this design.

Reply:
Sounds fair, I am OK with this with a follow-up Jira, thank you.